packages/block-editor/src/components/block-parent-selector/index.js

```diff
@@ -67,17 +73,38 @@ export default function BlockParentSelector() {
 		},
 	} );
 
+	// states to check reusable block edits
+	const [ reusableBlockHasEdits, setReusableBlockHasEdits ] = useState(
+		false
+	);
+
 	if ( shouldHide || firstParentClientId === undefined ) {
 		return null;
 	}
 
+	const classes = classnames( 'block-editor-block-parent-selector', {
+		'parent-block-has-changes': reusableBlockHasEdits,
+	} );
+
 	return (
 		<div
-			className="block-editor-block-parent-selector"
+			className={ classes }
 			key={ firstParentClientId }
 			ref={ nodeRef }
 			{ ...showMoversGestures }
 		>
+			{ /* Update the reusableBlockHasEdits state in BlockHasDot using Slot/Fill to update the parent-selector classes and to add dot to parent selector.  */ }
+			<Slot
+				name="parent-selector-has-dot"
+				fillProps={ {
+					setReusableBlockHasEdits,
+				} }
+			></Slot>
```

Comment:
> I don't think there's a need to use the slot to set a classname on the parent div. React components can be rendered in a slot, a bit like a portal.

The class is added to increase the `BlockParentSelector` width and to adjust its left position when the dot indicator is present. Since `BlockParentSelector` is positioned absolutely inside the block toolbar. 

Similar logic is used to increase the left margin of the `BlockContextualToolbar` to make room for the parent selector button when the dot indicator is present.

Is there some other way to conditionally load styles for `BlockContextualToolbar` and `BlockParentSelector` when the `BlockUnsavedChangesIndicator` is mounted?

Reply:
I was able to implement this using useEffect inside the `ParentBlockSelectorUnsavedChangesIndicator` component